src/Compilers/Core/Portable/Emit/EmitOptions.cs

```diff
@@ -225,6 +239,14 @@ internal void ValidateOptions(DiagnosticBag diagnostics, CommonMessageProvider m
                 diagnostics.Add(messageProvider.CreateDiagnostic(messageProvider.ERR_InvalidDebugInformationFormat, Location.None, (int)DebugInformationFormat));
             }
 
+            foreach (var instrumentationKind in InstrumentationKinds)
+            {
+                if (!instrumentationKind.IsValid())
+                {
+                    diagnostics.Add(messageProvider.CreateDiagnostic(messageProvider.ERR_InvalidInstrumentationKind, Location.None, (int)instrumentationKind));
```

Comment:
Clarifying the concept of instrumentation kind will also help me understand if/whether a numeric index is a meaningful way to represent it and communicate it to users. For instance, should an instrumentation kind have a number or an identifier/name?

Will this error message be useful? "Invalid instrumentation kind: 42"